### ISSUE TITLE
False positive for SC2199 in regex check

### ISSUE THREAD

**Issue by nachtgeist** [koalaman/shellcheck#3369]
#### For bugs
- SC2199
- v0.10.0
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/your/shebang
myArray=()
regex="fooregex"
if ! [[ "${inlined[@]}" =~ $regex ]]
then
  foo
fi
```

#### Here's what shellcheck currently says:
SC2199 (error): Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

#### Here's what I wanted or expected to see:
no output (false positive, since this is not an assignment but a regex check)

Thanks,
nachtgeist
